src/get-student-id.js

```diff
@@ -1,13 +1,12 @@
 import uuid from "uuid/v4";
 
-export const getStudentId = app =>
-  new Promise(resolve => {
-    app.userData.read("studentId", data => {
-      if (data && data.studentId) {
-        resolve(data.studentId);
-      } else {
-        data = { studentId: uuid() };
-        app.userData.write("studentId", data, () => resolve(data.studentId));
-      }
-    });
-  });
+const createNewId = dataStore => {
+  const id = uuid();
+  return dataStore.write("studentId", id).then(() => id, () => id);
+};
+
+export const getStudentId = dataStore =>
+  dataStore
+    .read("studentId")
+    .then(id => id || createNewId(dataStore))
+    .catch(() => createNewId(dataStore));
```

Comment:
So, this would trigger if the dataStore was unable to read from dataStore, right? If this happened but dataStore did contain studentId, would it mess things up to create a new ID?

I'm guessing you wrote it this way so that, at the end of the day, a studentId must be returned. That should be fine, just want to make sure we're considering the edge cases.

Reply:
Yeah it's a weird edge case but if we failed to read but were able to write, we'd be giving the student a new student ID. More than likely, though, failure to read also means failure to write. But like you said, we need an ID for the student to log events so it seemed reasonable to always create one, even if it means the student somehow accidentally changes IDs part way through.